README.md

```diff
@@ -421,6 +421,23 @@ val personId : messageId as Uid<Person>     // type to another using 'as'. This
                                             // in serialization.
 ```
 
+## Module: Function-Memoization
+
+Provides `memoize` extension to kotlin functions that allows optimizing expensive functions by 
+remembering the results corresponding to some set of specific inputs.
```

Comment:
Add a note that in order for this to work well, the output must only be a function of the specified arguments. Functions that take into account other inputs, such as 'the current time' or data from file files, or so, should never be memoized.
Also note that the equals and hashCode for the arguments of the functions must be well defined and obey the Kotlin hashCode/equals contract, or the caching may not work as espected.